### ISSUE TITLE
Unable to delete database when using redisdown.

### ISSUE THREAD

**Issue by gerbenupinion** [pouchdb/express-pouchdb#264]
In this setup I run express-pouchdb in combination with redisdown. I'm able to create, edit and delete documents. Creating a databases works as well. I'm only not able to delete a database. This causes the server to crash.

This issue appears when I run Redis on a different server then the pouchdb server. When I try to delete the dabatase it will try to connect to `127.0.0.1:6379` instead of the given `host` and `port`.

### Setting for redisdown.

```
opts.db = require('redisdown');
opts.port = 6379;
opts.host = 'redis.dev';
```

#### Request send:

```
DELETE /test HTTP/1.1
Host: redis.dev:80
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
Accept: application/json, text/javascript, */*; q=0.01
Origin: http://pouchdb.dev:80
X-Requested-With: XMLHttpRequest
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36
X-CouchDB-CSRF: true
Referer: http://pouchdb.dev:80/_utils/
```

#### Log:

```
[info] GET /test/_all_docs?limit=21 200 - 172.17.0.12
[info] GET /test/_all_docs?startkey=%22_design%2F%22&endkey=%22_design0%22&include_docs=true&limit=501 200 - 172.17.0.12
[info] GET /_all_dbs 200 - 172.17.0.12
[info] GET /_utils/js/zeroclipboard/ZeroClipboard.swf?noCache=1444901336410 200 - 172.17.0.12
[info] GET /_utils/img/couch-watermark.png 200 - 172.17.0.12

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED
    at errnoException (net.js:905:11)
    at Object.afterConnect [as oncomplete] (net.js:896:19)
```

**Comment by nolanlawson** [comment]
Yep, this looks like a bug in Redisdown rather than a bug in PouchDB. We heavily test a variety of *down adapters, including localstorage-down, memdown, fruitdown, and leveldown, but there are like dozens of these things, so we can't test them all unfortunately. :sweat_smile: 

Could you please file an issue over there? https://github.com/hmalphettes/redisdown

**Comment by nolanlawson** [comment]
Also you may want to mention your Node version and OS (windows/mac/linux).

**Comment by gerbenupinion** [comment]
Thank you for the quick response! I filed an issue in redisdown repository. https://github.com/hmalphettes/redisdown/issues/20

**Comment by gerbenupinion** [comment]
It seems to be a pouchdb `issue`.

When destroying a database any open connection is first closed and then reopened within the redisdown destroy function. This means the redisdown destroy function requires the same options param (host and port) as given when setting up a connection for a get or put request.

By default the leveldb adapter sends the params `location` and `callback` to the destroy function of a leveldown module. The redisdown module would work if the `options` param would be send as well.

Created a quick work around by creating  a LevelDOWN wrapper which adds the `options` param to the next request. It looks something like this:
```
    MonitoredLevelDOWN.destroy = function(location, callback) {
       var opts = {
         port = 6379,
         host = "redis.dev"
       }

       return wrappedClass.destroy(location, opts, callback);
    };
```

Hope this helps anybody.

**Comment by nolanlawson** [comment]
How the options object passed in to `destroy()`? Also a pull request on PouchDB would be most welcome: https://github.com/pouchdb/pouchdb/blob/b77788bb6d7964efb18a7d1be572eb27dc46af4a/lib/adapters/leveldb/index.js#L1440-L1461

**Comment by gerbenupinion** [comment]
Verify install becomes successful using redis when I change:

In https://github.com/pouchdb/pouchdb/blob/b77788bb6d7964efb18a7d1be572eb27dc46af4a/lib/adapters/leveldb/index.js#L159

`opts = utils.clone(opts);`
to
`var opts = utils.clone(opts);`

and https://github.com/pouchdb/pouchdb/blob/b77788bb6d7964efb18a7d1be572eb27dc46af4a/lib/adapters/leveldb/index.js#L1465

`leveldown.destroy(name, cb);`
to
`leveldown.destroy(name, opts, cb);`

But I'm afraid it will break when using other leveldown modules. For example MemDOWN. The destroy function expects param `name` && `callback`. See https://github.com/Level/memdown/blob/master/memdown.js#L223 in comparison to https://github.com/hmalphettes/redisdown/blob/master/index.js#L182.

**Comment by gerbenupinion** [comment]
@nolanlawson: I got a response from @hmalphettes in ticket https://github.com/hmalphettes/redisdown/issues/20. He is requesting the best practices regarding how pouchdb is calling the destroy methods of levelup. Maybe you could give some input on this subject?

Thxs

**Comment by nolanlawson** [comment]
Seems fixed in redisdown 0.1.9, please reopen if still an issue!
